src/repr/src/adt/datetime.rs

```diff
@@ -1938,14 +1929,35 @@ pub(crate) fn split_timestamp_string(value: &str) -> (&str, &str, CalendarEra) {
     }
 }
 
+// We support three era formats:
+//  1. "<timestamp><separator><timezone> <era>"
+//  2. "<timestamp> <era>"
+//  3. "<timestamp><era>"
+//
+// NB(ptravers): pg supports more formats than noted above.
 fn strip_era_from_timezone(timezone: &str) -> (&str, CalendarEra) {
     use CalendarEra::{AD, BC};
     let timezone = timezone.trim();
     let timezone_upper = timezone.to_uppercase();
 
+    // Covers cases 1 and 2.
+    if timezone.len() < 3 {
+        return match (
+            timezone_upper.strip_suffix("BC"),
+            timezone_upper.strip_suffix("AD"),
+        ) {
+            (Some(_), None) => ("", BC),
+            (None, Some(_)) => ("", AD),
+            // NB(ptravers): we expect this to fail when we go to check
+            // the timezone is valid in the next step.
+            _ => (timezone, AD),
+        };
+    }
+
+    // Covers case 3.
     match (
-        timezone_upper.strip_suffix(BC.as_str()),
-        timezone_upper.strip_suffix(AD.as_str()),
+        timezone_upper.strip_suffix(" BC"),
+        timezone_upper.strip_suffix(" AD"),
     ) {
         (Some(remainder), None) => (&timezone[..remainder.len()], BC),
         (None, Some(remainder)) => (&timezone[..remainder.len()], AD),
```

Comment:
It's not safe to index into a string based on a boundary obtained from another string. If the char boundaries are different it can cause a panic. The following commands cause the cluster to panic:

```sql
CREATE TABLE t1 (t text);
INSERT INTO t1 VALUES ('2025-01-01 00:00:00 Europe/Athensı AD');
SELECT t::timestamp FROM t1;
```

This is because the character `ı` is 2 bytes in lower case but one byte in upper case.

Reply:
Excellent point, will add that as a test

Reply:
done.